File: challenges/4-fizz-buzz/php/fabiosperotto/desafio-4.php
<?php
//checagem de input
if($argc < 2){
    echo "informe valor inteiro de entrada para o script\n";
    exit;
}
$limite = (int) $argv[1];
if($limite == 0){
    echo "informe um número válido\n";
    exit;
}

//algoritmo
for($i = 0; $i < $limite; $i++){
    $numero = $i + 1;
    if( ($numero % 3 == 0) && ($numero % 5 == 0) ){
        echo 'FizzBuzz';
    }else if($numero % 3 == 0){
        echo 'Fizz';
    }else if($numero % 5 == 0){
        echo 'Buzz';
    }else{
        echo $numero;
    }
    echo "\n";
}

File: challenges/4-fizz-buzz/go/KMarculo/desafio4.go
package main

import (
	"fmt"
)

func main() {
	var numMax int
	fmt.Printf("Digite um numero inteiro\n")
	fmt.Scan(&numMax)

	for i := 1; i <= numMax; i++ {
		if i%3 == 0 && i%5 == 0 {
			fmt.Printf("FizzBuzz\n")
		} else if i%3 == 0 {
			fmt.Printf("Fizz\n")
		} else if i%5 == 0 {
			fmt.Printf("Buzz\n")
		} else {
			fmt.Printf("%d\n", i)
		}
	}
}
